README.md

```diff
@@ -22,18 +18,132 @@ var bytes = MsgPackSerializer.Serialize(value);
+        {
+            var length = reader.ReadMapLength();
+            if (length == null)
+            {
+                return null;
+            }
+
+            if (length != 4)
+            {
+                throw new SerializationException("Bad format");
+            }
+
+            var result = new Beer();
+            for (var i = 0; i < length.Value; i++)
+            {
+                var propertyName = _stringConverter.Read(reader);
+                switch (propertyName)
+                {
+                    case nameof(result.Brand):
+                        result.Brand = _stringConverter.Read(reader);
+                        break;
+                    case nameof(result.Sort):
+                        result.Sort = _listStringConverter.Read(reader);
+                        break;
+                    case nameof(result.Alcohol):
+                        result.Alcohol = _floatConverter.Read(reader);
+                        break;
+                    case nameof(result.Brewery):
+                        result.Brewery = _stringConverter.Read(reader);
+                        break;
+                    default:
+                        throw new SerializationException("Bad format");
+                }
+            }
+
+            return result;
+        }
+
+        public void Initialize(MsgPackContext context)
+        {
+            _stringConverter = context.GetConverter<string>();
+            _listStringConverter = context.GetConverter<List<string>>();
+            _floatConverter = context.GetConverter<float>();
+            _context = context;
+        }
+    }
+```
+
+Then you should create create MsgPackContext and register your converter:
+```C#
+var context= new MsgPackContext();
+context.RegisterConverter(new BeerConverter());
+```
+
+And then you can serialize:
+```C#
+var bytes = MsgPackSerializer.Serialize(beer, context);
+```
+
+And deserialize:
+```C#
+var beer = MsgPackSerializer.Deserialize<Beer>(bytes, context);
 ```
 
 ## Performance
 * Serialization performance is comparable with msgpack.cli
 * Deserialization performance 2-3 times faster
-* MsgPack.Light works best if a data reside a memory (*_Array benchmarks).
-* Perfoming some IO operations, performance is suboptimal, but comparable with MsgPack.Cli (*_Stream benchmarks).
+* MsgPack.Light works best if a data reside a memory (\_*Array benchmarks).
+* Perfoming some IO operations, performance is suboptimal, but comparable with MsgPack.Cli (\*_Stream benchmarks).
 * More details can be found [here](https://github.com/roman-kozachenko/MsgPack.Light/blob/master/benchmarks.md).
 
 ## Credits
-* Benchmark data authors: https://github.com/aensidhe/SerializationPerformanceTest_CSharp and https://github.com/maximn/SerializationPerformanceTest_CSharp 
+* Benchmark data copied from [thekvs cpp-serializers project](https://github.com/thekvs/cpp-serializers/blob/c6b305fe3659d2df3b492698bb5d7cb284ab2f9f/data.hpp).
```

Comment:
This one should get credit too: https://github.com/maximn/SerializationPerformanceTest_CSharp